docs/hub/webhooks-guide-metadata-review.md

```diff
@@ -0,0 +1,207 @@
+
+Since we have a Python dictionary containing our metadata, we can loop through this dictionary to check if our desired keys are there. If a desired metadata field (a key in our dictionary) is missing, we'll assign the value as `None`.
+
+```python
+def create_metadata_key_dict(card_data, repo_type: str):
+    shared_keys = ["tags", "license"]
+    if repo_type == "model":
+        model_keys = ["library_name", "datasets", "metrics", "co2", "pipeline_tag"]
+        shared_keys.extend(model_keys)
+        keys = shared_keys
+        return {key: card_data.get(key) for key in keys}
+    if repo_type == "dataset":
+        # [...]
+```
+
+This function will return a dictionary containing keys representing the metadata fields we require for our model or dataset. The dictionary values will either include the metadata entered for that field or `None` if that metadata field is missing in the `YAML`. 
+
+```python
+{'tags': None,
+ 'license': 'afl-3.0',
+ 'library_name': None,
+ 'datasets': None,
+ 'metrics': None,
+ 'co2': None,
+ 'pipeline_tag': None}
+```
+
+Once we have this dictionary, we can create our metadata report. In the interest of brevity, we won't include the complete code here, but the Hugging Face Spaces [repository](https://huggingface.co/spaces/librarian-bot/webhook_metadata_reviewer/blob/main/main.py) for this webhook contains the full code.
+
+We create one function which creates a markdown table that produces a prettier version of the data we have in our metadata coverage dictionary. 
+
+```python
+def create_metadata_breakdown_table(desired_metadata_dictionary):
+    # [...]
+    return tabulate(
+        table_data, tablefmt="github", headers=("Metadata Field", "Provided Value")
+    )
+```
+
+We also have a Python function that generates a score (representing the percentage of the desired metadata fields present)
+
+```python
+def calculate_grade(desired_metadata_dictionary):
+    # [...]
+    return round(score, 2)
+```
+
+and a Python function that creates a markdown report for our metadata review. This report contains both the score and metadata table, along with some explanation of what the report contains.
+
+
+```python
+def create_markdown_report(
+    desired_metadata_dictionary, repo_name, repo_type, score, update: bool = False
+):
+    # [...]
+    return report
+```
+
+
+## How to post the review automatically?
+
+We now have a markdown formatted metadata review report. We'll use the `huggingface_hub` library to post this review. We define a function that takes back the webhook data received from the Hub, parses the data, and creates the metadata report. Depending on whether a report has previously been created, the function creates a new report or posts a new issue to an existing metadata review thread.
+
+```python
+def create_or_update_report(data):
+    if parsed_post := parse_webhook_post(data):
```

Comment:
maybe mention this is very cool, but a new feature in Python (and maybe mention the name of the syntax)

as not everyone knows this IMO (but i love it)